apps/api/src/app/workflows-v2/usecases/extract-variables/extract-variables.usecase.ts

```diff
@@ -68,71 +71,20 @@ export class BuildPayloadSchema {
     for (const [key, value] of Object.entries(controlValue)) {
       if (isStringTipTapNode(value)) {
         processedValue[key] = this.hydrateEmailSchemaUseCase.execute({ emailEditor: value });
+      } else if (key === 'skip' && isValidRule(value as RulesLogic<AdditionalOperation>)) {
+        const fields = extractFieldsFromRules(value as RulesLogic<AdditionalOperation>)
+          .filter((field) => field.startsWith('payload.') || field.startsWith('subscriber.data.'))
+          .map((field) => `{{${field}}}`);
+
+        processedValue[key] = {
+          rules: value,
+          fields,
+        };
```

Comment:
Extract the `payload.` and `subscriber.data.` fields from the step condition rules. Later they are used to build the variables and it's schema.